futures-util/src/stream/select_with_strategy.rs

```diff
@@ -170,47 +199,88 @@ where
     Clos: FnMut(&mut State) -> PollNext,
 {
     fn is_terminated(&self) -> bool {
-        self.stream1.is_terminated() && self.stream2.is_terminated()
+        match self.internal_state {
+            InternalState::BothFinished => true,
+            _ => false,
+        }
     }
 }
 
-impl<St1, St2, Clos, State> Stream for SelectWithStrategy<St1, St2, Clos, State>
+#[inline(always)]
```

Comment:
Why is this #[inline(always)]? AFAIK #[inline] is enough in most cases.

Reply:
You're right, it looks like the binaries produced are identical.
I'll change it to `#[inline]`.